torch/quantization/observer.py

```diff
@@ -45,15 +53,47 @@ def forward(self, x):
     def calculate_qparams(self, **kwargs):
         pass
 
+    def _calculate_per_channel_qparams(self, min_vals, max_vals):
+        # type: (Optional[Tensor], Optional[Tensor]) -> Tuple[Tensor, Tensor]
+        """
+        Given min and max value tensors, this function calculates per channel
+        quantization parameters
+        """
+        if min_vals is None or max_vals is None:
+            warnings.warn(
+                "must run observer before calling calculate_qparams.\
+                                    Returning default scale and zero point "
+            )
+            return torch.tensor([1.0]), torch.tensor([0])
+
+        for i in range(len(min_vals)):
+            assert (
+                min_vals[i] <= max_vals[i]
+            ), "min {} should be less than max {}".format(min_vals[i], max_vals[i])
+
+        scales = torch.ones(min_vals.size())
+        zero_points = torch.ones(min_vals.size())
+        for i in range(len(scales)):
+            # scales[i], zero_points[i] = self._calculate_qparams(
+            qparam = self._calculate_qparams(
+                min_vals[i], max_vals[i]
+            )
+            scales[i] = float(qparam[0])
```

Comment:
Do you need to convert to float/int here? Both qparam and scales are tensors

Reply:
Without float/int, scriptable test would have error:
> RuntimeError: output with shape [] doesn't match the broadcast shape [1]
probably because scales[i] is the scalar.